### PR TITLE
Members only

### PR DESCRIPTION
**Pull request by jacobo12montes**
-Definde the models to be use on the project
-Create the methods in the controllers for each model
-Define the relation between the objects
-Add the logic to allow see the author only to the members

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class = "center" >
+<h4><%= @post.title %></h4>
+<p><%= @post.body %></p>
+<% if current_member %>
+          <p>Author: <%= post.member.name %></p>
+        <% end %><%= link_to "<< Back to Post List", posts_path %>
+<%= link_to "Create a New Post", new_post_path, class: "new_post" %>
```
Comment (bertil291utn) [inline app/views/posts/show.html.erb:7]: ## Add @ 
- Post is a class variable coming from the controller therefore add `@`
```ruby
  <p>Author: <%= @post.member.name %></p>
```

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Rails.application.routes.draw do
+  devise_for :members
+  resources :posts
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
```
Comment (bertil291utn) [inline config/routes.rb:4]: ## Post routes
- Create the application routes using `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
  - [x] Replace `resources :posts, only: [:new, :create, :index]`, it's better to add specific behaviour to each route

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it Git Flow you have a professional README file. 

However, in order to accomplish with all requirements , you required to do these changes, listed below:
### MANDATORY 

- The application has Sign in/Sign out links present in any view of the app ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))  
  - [x] To approach this section to add a fixed navbar at the page top 
- [x] Please add some styles to the web page

### OPTIONAL
- [x] When user login add a link to see the posts, so far it's redirecting to create direct a post 
- [x] Add a workflow the first  page that user see is the posts


### REMINDER
- It was an issue with the migrations, I had to comment on the two devise [migrations](https://files.slack.com/files-pri/TJG70AZ1N-F01CBA0Q9NK/memebers-only-rails.gif),

  - Next time test the migrations with a clean db to make sure they work and remove migrations.

### Summary
- Sign in/Sign out links should be on all pages
- Add style to the web page to have a better workflow
-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

**Comment by jacobo12montes** [comment]
-We have changed the layout application
-Add login and sign up to every view
-Add style to the app

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 
## Great job guys :clap: :clap: :clap:, thanks for making the required changes by the previous TSE.

now you can merge, your project is approved.

## Happy Coding :tophat:

**Pull request merged**